akka-persistence/src/main/scala/akka/persistence/AtLeastOnceDelivery.scala

```diff
@@ -380,6 +384,8 @@ abstract class AbstractPersistentActorWithAtLeastOnceDelivery extends AbstractPe
    * This method will throw [[AtLeastOnceDelivery.MaxUnconfirmedMessagesExceededException]]
    * if [[numberOfUnconfirmed]] is greater than or equal to [[#maxUnconfirmedMessages]].
    */
-  def deliver(destination: ActorPath, deliveryIdToMessage: akka.japi.Function[java.lang.Long, Object]): Unit =
+  def deliver(destination: ActorPath, deliveryIdToMessage: akka.japi.function.Function[java.lang.Long, Object]): Unit =
     super.deliver(destination, id ⇒ deliveryIdToMessage.apply(id))
+
+  // FIXME @deprecated deliver with akka.japi.Function. How would java8 know which one to pick when param is lambda?
```

Comment:
I think it makes sense to have a complete set of these functions in `akka.japi.function`, even if they duplicate things (with slight variation) that exist in `akka.japi`. We may break compatibility in Akka 3.0, but for now we could just punt and do nothing, i.e. back out the changes that use the new functions where problematic (since that usage does not gain us anything besides internal consistency anyway). Then we can talk with Josh _et al_ about how we could tackle such a change.